src/main/java/com/r307/arbitrader/service/TradingService.java

```diff
@@ -470,15 +471,22 @@ private BigDecimal addFees(Exchange exchange, CurrencyPair currencyPair, BigDeci
                 .multiply(exchangeService.getExchangeFee(exchange, currencyPair, true))
                 .setScale(BTC_SCALE, RoundingMode.HALF_EVEN);
 
-            BigDecimal adjustedVolume = volume.add(fee);
+            final CurrencyPairMetaData currencyPairMetaData = exchange
```

Comment:
This doesn't seem right to do `roundByStep()` inside `addFees()`. We do it once in `getVolumeForEntryPosition()` and then possibly `addFees()` changes the value. I think instead we should move the `roundByStep()` out into `entryPosition()` after `addFees()` so that we do the rounding as the last step, in one place, right before submitting the orders. Same idea for exiting the positions.

Reply:
Yeah, good point. I will commit the changes in a moment